### DISCUSSION TITLE
Contact with explicit time integrator

### DISCUSSION THREAD

**Discussion by lyyc199586** [idaholab/moose#25401] [Q&A Modules: Solid mechanics] [answered]
Can we use contact with explicit time integrator (e.g. CentralDifference)?
I tried with the example in introduction of contact modules (https://mooseframework.inl.gov/modules/contact/tutorials/introduction/step01.html), with

```
[Executioner]
  type = Transient
  end_time = 5
  dt = 0.5
  [TimeIntegrator]
    type = CentralDifference
  []
[]
```

but only get
```
*** ERROR ***
Need a system matrix 
```

**Comment by GiudGiud** [comment]
Hello

CentralDifference is implemented a little differently. Do you mind trying the other explicit time integrators?

Guillaume

**Comment by lindsayad** [comment]
If you are interested in doing dynamics, I believe [NewmarkBeta](https://mooseframework.inl.gov/source/timeintegrators/NewmarkBeta.html) is often used. @recuero might want to comment more

**Comment by recuero** [comment, marked as answer]
Contact has not been implemented/tested for explicit dynamics (we will hopefully get to that). For implicit dynamics, use of contact objects is limited, but we do have examples or tests that use Newmark beta.

**Comment by lyyc199586** [comment]
got it, thanks

**Reply by recuero** [reply]
FYI: We are going to be looking into explicit dynamics contact this year: https://github.com/idaholab/moose/issues/25666

**Reply by lyyc199586** [reply]
great!
